impeller/geometry/path.cc

```diff
@@ -64,51 +65,47 @@ void Path::SetConvexity(Convexity value) {
 }
 
 void Path::Shift(Point shift) {
-  size_t currentIndex = 0;
-  for (const auto& component : components_) {
-    switch (component.type) {
-      case ComponentType::kLinear:
-        linears_[component.index].p1 += shift;
-        linears_[component.index].p2 += shift;
-        break;
-      case ComponentType::kQuadratic:
-        quads_[component.index].cp += shift;
-        quads_[component.index].p1 += shift;
-        quads_[component.index].p2 += shift;
-        break;
-      case ComponentType::kCubic:
-        cubics_[component.index].cp1 += shift;
-        cubics_[component.index].cp2 += shift;
-        cubics_[component.index].p1 += shift;
-        cubics_[component.index].p2 += shift;
-        break;
-      case ComponentType::kContour:
-        contours_[component.index].destination += shift;
-        break;
-    }
-    currentIndex++;
+  for (auto i = 0u; i < points_.size(); i++) {
```

Comment:
FYI this is where having a linear algrebra library would be nice.  This sort of operation would be 10x faster if it was vectorized.

Reply:
This isn't really a bottleneck though!